File: backend/schemas/recommendation.py
"""
Job Intelligent - Recommendation Schemas
Pydantic models for recommendation responses.
"""
from pydantic import BaseModel
from backend.schemas.job import JobOut


class RecommendationOut(BaseModel):
    """A recommended job with a relevance score."""
    job: JobOut
    score: float
    match_type: str  # "keyword" or "semantic"

File: lib/domain/entities/recommendation.dart
/// Job Intelligent - Recommendation Entity
library;

import 'job.dart';

class RecommendationEntity {
  final JobEntity job;
  final double score;
  final String matchType;

  const RecommendationEntity({
    required this.job,
    required this.score,
    required this.matchType,
  });
}
